cmd/sso.go

```diff
@@ -0,0 +1,202 @@
+package cmd
+
+import (
+	"errors"
+	"os"
+	"os/signal"
+	"strings"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+
+	"github.com/supabase/cli/internal/sso/create"
+	"github.com/supabase/cli/internal/sso/get"
+	"github.com/supabase/cli/internal/sso/info"
+	"github.com/supabase/cli/internal/sso/list"
+	"github.com/supabase/cli/internal/sso/remove"
+	"github.com/supabase/cli/internal/sso/update"
+	"github.com/supabase/cli/internal/utils"
+)
+
+var (
+	ssoCmd = &cobra.Command{
+		GroupID: groupManagementAPI,
+		Use:     "sso",
+		Short:   "Manage Single Sign-On (SSO) authentication for projects",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := cmd.Root().PersistentPreRunE(cmd, args); err != nil {
+				return err
+			}
+			ctx, _ := signal.NotifyContext(cmd.Context(), os.Interrupt)
+			cmd.SetContext(ctx)
+
+			fsys := afero.NewOsFs()
+			if err := PromptProjectRef(fsys, cmd); err != nil {
+				return err
+			}
+
+			// Validate project ref
+			if !utils.ProjectRefPattern.MatchString(projectRef) {
+				return errors.New("Invalid project ref format. Must be like `abcdefghijklmnopqrst`.")
+			}
+			return nil
+		},
+	}
+
+	ssoMetadataFile         string
+	ssoMetadataURL          string
+	ssoSkipURLValidation    bool
+	ssoMetadata             bool
+	ssoAttributeMappingFile string
+	ssoDomains              []string
+	ssoAddDomains           []string
+	ssoRemoveDomains        []string
+	ssoOutput               = utils.EnumFlag{
+		Allowed: utils.OutputDefaultAllowed,
+		Value:   utils.OutputPretty,
+	}
+
+	ssoAddCmd = &cobra.Command{
+		Use:     "add <type = saml> [flags]",
+		Short:   "Add a new SSO identity provider",
+		Args:    cobra.ExactArgs(1),
+		Example: `  supabase sso add saml --project-ref mwjylndxudmiehsxhmmz --metadata-file ~/SAMLMetadata.xml --domains example.com`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if strings.ToLower(args[0]) != "saml" {
+				return errors.New("Identity provider type must be \"saml\"")
+			}
+
+			return create.Run(cmd.Context(), create.RunParams{
+				ProjectRef: projectRef,
+				Format:     ssoOutput.Value,
+
+				MetadataFile:      ssoMetadataFile,
+				MetadataURL:       ssoMetadataURL,
+				SkipURLValidation: ssoSkipURLValidation,
+				AttributeMapping:  ssoAttributeMappingFile,
+				Domains:           ssoDomains,
+			})
+		},
+	}
+
+	ssoRemoveCmd = &cobra.Command{
+		Use:     "remove <provider-id> [flags]",
+		Short:   "Remove an existing SSO identity provider",
+		Args:    cobra.ExactArgs(1),
+		Example: `  supabase sso remove b5ae62f9-ef1d-4f11-a02b-731c8bbb11e8 --project-ref mwjylndxudmiehsxhmmz`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return remove.Run(cmd.Context(), projectRef, args[0], ssoOutput.Value)
```

Comment:
Is it possible to validate the providerId with regex?, ie. `args[0]`

Reply:
yeah should be possible since it follows the uuid_v4 format